sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableServiceImpl.java

```diff
@@ -93,12 +93,32 @@ class BigtableServiceImpl implements BigtableService {
   private static final long MIN_BYTE_BUFFER_SIZE = 100 * 1024 * 1024; // 100MB
 
   public BigtableServiceImpl(BigtableDataSettings settings) throws IOException {
-    this.client = BigtableDataClient.create(settings);
+    this(settings, null);
+  }
+
+  public BigtableServiceImpl(BigtableDataSettings settings, Duration readWaitTimeout)
+      throws IOException {
     this.projectId = settings.getProjectId();
     this.instanceId = settings.getInstanceId();
     RetrySettings retry = settings.getStubSettings().readRowsSettings().getRetrySettings();
     this.readAttemptTimeout = Duration.millis(retry.getInitialRpcTimeout().toMillis());
     this.readOperationTimeout = Duration.millis(retry.getTotalTimeout().toMillis());
+    BigtableDataSettings.Builder builder = settings.toBuilder();
+    if (readWaitTimeout != null) {
+      // TODO: a hack to workaround incorrect mapping from attempt timeout to watchdogs wait
+      // timeout. This can be removed once
+      // https://github.com/googleapis/gapic-generator-java/pull/1473 is resolved.
```

Comment:
This chunk should be done along with assembling BigtableDataSettings before pass to the constructor of BigtableServiceImpl, then we do not need to add an temporary (from my understanding to this TODO) argument to the public BigtableServiceImpl constructor. Removing it later will be a breaking change.

Reply:
This class is package private, would it still be a breaking change? I also made the constructors package private.

Reply:
ah I see the need of taking another `readWaitTimeout` now. Thanks for clarification. Then probably add "TODO remove this overloaded contructor once ... is resolved" on top of `BigtableServiceImpl(BigtableDataSettings, Duration)` method?

Reply:
updated!

Reply:
Thanks!